curriculum/challenges/english/25-front-end-development/lecture-working-with-loops-and-sequences/6839e3ea328b630cf5fca9ac.md

```diff
@@ -1,131 +1,292 @@
+
+```python
+words = ['sky', 'apple', 'rhythm', 'fly', 'orange']
+
+for word in words:
+    for letter in word:
+        if letter.lower() in 'aeiou':
+            print(f"'{word}' contains the vowel '{letter}'")
+            break
+    else:
+        print(f"'{word}' has no vowels")
+```
+
+In this example we have a list of random words, and a `for` loop is used to loop through each word. Inside the outer `for` loop, we have another `for` loop to loop through each letter of each word. If the lowercase version of the letter is a vowel, we print the word followed by what vowels it contains, then break out of the inner loop. If the word contains no values, then we print a message indicating that.
+
+Here is what the result looks like in the console:
+
+```md
+'sky' has no vowels
+'apple' contains the vowel 'a'
+'rhythm' has no vowels
+'fly' has no vowels
+'orange' contains the vowel 'o'
+```
+
+Loops are very common in Python, so it's important to get comfortable with them. In the next few lectures, you will learn how to work with the `enumerate()` and `range()` functions in loops.
 
 # --questions--
 
 ## --text--
 
-What will the following code return?
+What will the following code print to the console?
 
 ```py
-programming_languages = ('Rust', 'Java', 'Python', 'C++', 'Rust')
-programming_languages.count('Rust')
+programming_languages = ['Rust', 'Java', 'Python', 'C++']
+
+for language in programming_languages:
+    print(language)
 ```
 
 ## --answers--
 
-`2`
+```md
+'Rust'
+'Java'
+'Python'
+'C++'
+```
 
 ---
 
-`1`
+```md
+'Rust'
+'Python'
+'C++'
+```
 
 ### --feedback--
 
-Review the beginning of the video for the answer.
+Review the beginning of the lecture for the answer.
 
 ---
 
-`0`
+```md
+'Rust'
+'Java'
+```
 
 ### --feedback--
 
-Review the beginning of the video for the answer.
+Review the beginning of the lecture for the answer.
 
 ---
 
-`3`
+```md
+'Rust'
+'Python'
+'C++'
+```
 
 ### --feedback--
 
-Review the beginning of the video for the answer.
+Review the beginning of the lecture for the answer.
 
 ## --video-solution--
 
 1
 
 ## --text--
 
-What will be the result for the following code?
-
-```py
-programming_languages = ('Rust', 'Java', 'Python', 'C++', 'Rust')
-programming_languages.index('JavaScript')
-```
+Which of the following loops is used to repeat a block of code until a condition is `False`?
 
 ## --answers--
 
-`IndexError`
+`list`
 
 ### --feedback--
 
-Think about which type of error is appropriate to communicate that a value cannot be found.
+Only one of these options is a real type of loop used in both Python and JavaScript.
 
 ---
 
-`RangeError`
+`while`
 
-### --feedback--
+---
 
-Think about which type of error is appropriate to communicate that a value cannot be found.
+`continue`
 
----
+### --feedback--
 
-`ValueError`
+Only one of these options is a real type of loop used in both Python and JavaScript.
```

Comment:
the format is correct, this diff mess here is that which answer is the correct one is different in the new question